### PR TITLE
Ar/adding identity to livepeer.com ui tracking

### PR DESCRIPTION
**Pull request by angyangie**
**What does this pull request do? Explain your changes. (required)**
https://github.com/livepeer/livepeerjs/issues/703
https://github.com/livepeer/livepeerjs/issues/705

**Specific updates (required)**
- Adds identity to every page view when a user signs up, which associates anonymous records with specific `emails` / `userIds` that can then be used to aggregate all interactions with Livepeer a user might have much more accurately. This info can then be used to pump data back into Hubspot or any other service.
- Tracks referral information
- Removes HubSpot tracking code to prevent duplicates

**How did you test each of these updates (required)**
Segment.com and Hubspot.com

**Does this pull request close any open issues?**
https://github.com/livepeer/livepeerjs/issues/703
https://github.com/livepeer/livepeerjs/issues/705

### CODE REVIEW

**Comment by angyangie** [comment]
Confirmed with Mark that it is safe to remove HubSpot code and lose the HubSpot page view records going forward. They will live in Segment.

File: packages/api/src/controllers/helpers.js [hunk 1]
```diff
@@ -144,3 +145,27 @@ export async function sendgridEmail({
...
+    traits: {
+      email: email,
+    },
+  })
+
+  const properties = {}
+  if ('properties' in event) {
+    for (const key in Object.keys(event.properties)) {
+      properties[key] = event.properties[key]
+    }
```
Comment (iameli) [inline packages/api/src/controllers/helpers.js]: Easier way to write this:
```suggestion
    properties = {...properties, ...event.properties}
```

File: packages/api/src/controllers/helpers.js [hunk 2]
```diff
@@ -144,3 +145,27 @@ export async function sendgridEmail({
   SendgridMail.setApiKey(sendgridApiKey)
   await SendgridMail.send(msg)
 }
+
+export async function trackAction(userId, email, event, segmentApiKey) {
+  var analytics = new SegmentAnalytics(segmentApiKey, { flushAt: 1 })
```
Comment (iameli) [inline packages/api/src/controllers/helpers.js:153]: Do nothing if no segmentApiKey

**Review by iameli — APPROVED** [review]
Let's just make sure we don't track if we don't have a segment code and whatnot

**Pull request merged**